core/Plugin/Metric.php

```diff
@@ -119,18 +119,30 @@ public static function getMetric($row, $columnName, $mappingNameToId = null)
             ) {
                 $value = $row->getColumn($mappingNameToId[$columnName]);
             }
-        } else {
-            $value = @$row[$columnName];
-            if ($value === null
-                && isset($mappingNameToId[$columnName])
-            ) {
-                $columnName = $mappingNameToId[$columnName];
-                $value = @$row[$columnName];
-            }
+
             return $value;
+
+        } elseif (!empty($row)) {
+            
+            if (array_key_exists($columnName, $row)) {
+                return $row[$columnName];
+            } else {
+
+                if (empty($mappingNameToId)) {
```

Comment:
technically if you move this `if` block here, then the block above L111 should also be moved in the first `if` statement